File: geocoder/src/main/java/com/prolificinteractive/geocoder/Downloader.java
package com.prolificinteractive.geocoder;

import io.reactivex.disposables.Disposable;
import java.io.IOException;

public interface Downloader {
  Disposable getDisposable();

  byte[] request(final String url) throws IOException;

  abstract class Factory {
    public abstract Downloader create();
  }
}

File: geocoder/src/main/java/com/prolificinteractive/geocoder/Downloader.kt
package com.prolificinteractive.geocoder

import io.reactivex.disposables.Disposable
import java.io.IOException

interface Downloader {
  val disposable: Disposable

  @Throws(IOException::class)
  fun request(url: String): ByteArray

  interface Factory {
    fun create(): Downloader
  }
}